Sources/Transaction/Memos/SenderWithPaymentRequestMemoUtils.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+import Foundation
+import LibMobileCoin
+
+enum SenderWithPaymentRequestMemoUtils {
+
+    static func isValid(
+        memoData: Data64,
+        senderPublicAddress: PublicAddress,
+        receipientViewPrivateKey: RistrettoPrivate,
+        txOutPublicKey: RistrettoPublic
+    ) -> Bool {
+        memoData.asMcBuffer { memoDataPtr in
+            senderPublicAddress.withUnsafeCStructPointer { publicAddressPtr in
+                receipientViewPrivateKey.asMcBuffer { receipientViewPrivateKeyPtr in
+                    txOutPublicKey.asMcBuffer { txOutPublicKeyPtr in
+                        var matches = true
```

Comment:
We should probably default matches to false here...shouldn't change things if mc_memo_sender_with_payment_request_memo_is_valid behaves properly and always sets it on .success, but just seems better to default to false when looking for a match in case of a bug in the called function.